recipes/wasmtime/all/conanfile.py

```diff
@@ -34,18 +33,14 @@ def configure(self):
         if self.options.shared:
             del self.options.fPIC
 
-    def build(self):
-        try:
-            if self.settings.arch == "armv8" and self.settings.os == "Android":
-                os_name = "Linux"
-            else:
-                os_name = str(self.settings.os)
+    def validate(self):
+        if (not (self.version in self.conan_data["sources"]) or
+            not (str(self.settings.os) in self.conan_data["sources"][self.version]) or
+            not (str(self.settings.arch) in self.conan_data["sources"][self.version][str(self.settings.os)] ) ):
+            raise ConanInvalidConfiguration("Binaries for this combination of architecture/version/os not available")
 
-            archive_ext = "zip" if os_name == "Windows" else "tar.xz"
-            url = f"https://github.com/bytecodealliance/wasmtime/releases/download/v{self.version}/wasmtime-v{self.version}-{self.settings.arch}-{os_name.lower()}-c-api.{archive_ext}"
-            tools.get(url, strip_root=True, destination=self._source_subfolder)
-        except:
-            raise Exception("Binary does not exist for these settings")
+    def build(self):
```

Comment:
```suggestion
    def source(self):
```

Reply:
After this change I got this error:
```console
ERROR: wasmtime/0.28.0 (test package): Error in build() method, line 11
        cmake.configure()
```

Reply:
Then the extraction or packaging is in correct.

Downloading sources should be in the source method not build

Reply:
Done